### PR TITLE
cicd: openshift CI

### PR DESCRIPTION
**Pull request by hdonnay**
This retools the Makefile and Dockerfile to use a buildkit backend directly in an attempt to stop needing to write a Dockerfile and scripts that only use the intersection of `podman` and `docker` features.

TODO:
- [x] Documentation updates
- [x] ~~Port the GitHub Actions CI to use the Makefile where appropriate.~~ (Will do in additional PRs)

### CODE REVIEW

**Comment by hdonnay** [comment]
Currently failing due to extremely annoying AppSRE footguns, will pick up later.

**Comment by hdonnay** [comment]
cc @Marcusk19 @bcaton85

There might be some Quay build concerns, so FYI.

File: etc/config.mk [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# The package used (via `go run`) to format go files.
+goimports ?= golang.org/x/tools/cmd/goimports@latest
+
+# These are arguments added to `go test` invocations.
+testargs =
+
+# VERSION is the version used when guessing at the current version to be used
+# in dist artifacts.
+#
+# This should be able to be interpreted accorting to gitrevisions(7) and a
```
Comment (crozzy) [inline etc/config.mk]: nit: spelling

File: etc/config.mk [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# `docker` and `docker-compose` control the commands invoked for a container
+# engine and a "compose file" handler, respectively.
+docker ?= $(notdir $(shell command -v podman 2>/dev/null || command -v docker 2>/dev/null))
+ifndef docker
+$(error 'docker' must not be defined as empty)
```
Comment (crozzy) [inline etc/config.mk]: Do these errors get surfaced to the user? If so should we mention that 'podman' will suffice too?

File: etc/container.mk [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ifdef DEBUG
+buildctl += --progress plain
+endif
+buildctl += --opt build-arg:BUILDKIT_MULTI_PLATFORM=1
+buildctl += $(strip $(foreach v,$(buildkit_passthru),$(if $($v),--opt build-arg:$v=$($v),)))
+buildctl += --local context=$$src
+buildctl += --local dockerfile=$$src
+buildctl += --output $(call splice,$(output_args))
+ifdef GITHUB_ACTIONS
+buildctl += --export-cache type=gha,mode=max
```
Comment (crozzy) [inline etc/container.mk:17]: huh, special mode for gh actions, nice

**Review by crozzy — CHANGES_REQUESTED** [review]

File: etc/config.mk (hunk 2 above)
Comment (hdonnay) [reply etc/config.mk]: They do, I'll make that change

**Comment by hdonnay** [comment]
I guess ignore the devshift failure, despite it being the motivator of this whole exercise?

**Comment by hdonnay** [comment]
[test]

**Review by crozzy — DISMISSED** [review]
SGTM

**Review by crozzy — APPROVED** [review]

**Pull request merged**